### DISCUSSION TITLE
Powerlevel10k instant prompt

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [marlonrichert/zsh-snap#133] [Q&A] [answered]
I use Powerlevel10k and for instant prompt I copied and pasted this from it's website. It works fine no issues. I wondering if there's a more "zsh-snapish" way of achieving the same result? Somthing like `znap prompt romkatv/powerlevel10k` 
perhaps?

```
# Enable Powerlevel10k instant prompt. Should stay close to the top of ~/.zshrc.
# Initialization code that may require console input (password prompts, [y/n]
# confirmations, etc.) must go above this block; everything else may go below.
if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi
```

**Comment by marlonrichert** [comment, marked as answer]
No, not really. P10k generates that file itself and, apparently, gives no guarantee that the file will be there at any given point in time.

You could refactor the code a little, though:
```zsh
() { [[ -r $1 ]] && source $1 } ${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-$USERNAME.zsh
```

**Comment by whi-tw** [comment]
In case anyone else stumbles upon this, I've found a 'solution' for this which both ensures `znap` continues managing the p10k install (so it can perform updates) and allows p10k's instant prompt to keep working:

```zsh
source ~/.znap/zsh-snap/znap.zsh

run-all
your-noisy
commands-here

() { [[ -r $1 ]] && source $1 } ${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${USERNAME}.zsh

znap command
znap command
other-things

# put / source your p10k config here

## This should be on the last line
znap source romkatv/powerlevel10k powerlevel10k.zsh-theme
```

This ensures that the instant prompt is sourced early on, and then hands over to znap to load the theme via `znap source`.

Replacing the `znap source` with `znap prompt romkatv/powerlevel10k` does 'work' but as there is some output to stdout, p10k complains when `${POWERLEVEL9K_INSTANT_PROMPT}` is set to `verbose`. The prompt also jumps around a bit which is annoying.

For my actual `.zshrc`, see: https://github.com/whi-tw/.zshrc (files are written to ~/.zshrc in numerical order).
